pkg/nodes/queue.go

```diff
@@ -0,0 +1,95 @@
+package nodes
+
+import (
+	"context"
+	"time"
+
+	"github.com/celestiaorg/torch/config"
+	"github.com/celestiaorg/torch/pkg/db/redis"
+	"github.com/celestiaorg/torch/pkg/k8s"
+	"github.com/celestiaorg/torch/pkg/metrics"
+
+	log "github.com/sirupsen/logrus"
+)
+
+var (
+	taskQueue     = make(chan config.Peer) // taskQueue channel for pending tasks (peers to process later).
+	MaxRetryCount = 5                      // MaxRetryCount number of retries per node.
+)
+
+// ProcessTaskQueue processes the pending tasks in the queue every 5 seconds.
+func ProcessTaskQueue() {
+	ticker := time.NewTicker(5 * time.Second) // Set the interval to 5 seconds
+	//defer ticker.Stop()                       // Stop the ticker when the function exits
+
+	for {
+		select {
+		case <-ticker.C:
+			processQueue()
+		}
+	}
+}
+
+// processQueue process the nodes in the queue and tries to generate the Multi Address
+func processQueue() {
+	red := redis.InitRedisConfig()
+	ctx := context.TODO()
```

Comment:
As a good practice you need to pass context downstream so you can have a select case if the context is either done or not